gostream/ffmpeg/avframe.go

```diff
@@ -1,30 +1,23 @@
-//go:build cgo && ((linux && !android) || (darwin && arm64))
-
-// Package avutil is a wrapper around FFmpeg/release6.1.
-// See: https://github.com/FFmpeg/FFmpeg/tree/release/6.1
-package avutil
-
-//#cgo CFLAGS: -I${SRCDIR}/../include
-//#cgo linux,arm64 LDFLAGS: -L${SRCDIR}/../Linux-aarch64/lib -lavformat -lavcodec -lavutil -lm
-//#cgo linux,amd64 LDFLAGS: -L${SRCDIR}/../Linux-x86_64/lib -lavformat -lavcodec -lavutil -lm
-//#cgo linux,arm LDFLAGS: -L${SRCDIR}/../Linux-armv7l/lib -lavformat -lavcodec -lavutil -lm
-//#cgo darwin,arm64 LDFLAGS: -L${SRCDIR}/../Darwin-arm64/lib -lavformat -lavcodec -lavutil -lm
-//#include <libswresample/swresample.h>
-//#include <libavutil/error.h>
-//#include <stdlib.h>
-// static const char *error2string(int code) { return av_err2str(code); }
+//go:build cgo && !android
+
+package ffmpeg
+
+//#include <libavutil/frame.h>
+//#include <libavutil/imgutils.h>
 import "C"
 
 import (
 	"image"
-	"reflect"
 	"unsafe"
 
 	"github.com/pkg/errors"
 )
 
 // Frame an AVFrame
-type Frame C.struct_AVFrame
+type Frame struct {
+	p  *C.struct_AVFrame
+	pp **C.struct_AVFrame
```

Comment:
[q] Is `pp` only needed for freeing?

Reply:
Well, anywhere a double pointer is required by the C functions. Trying to use `&p` gave me segfaults when I attempted to free the AVDictionary.